### PR TITLE
Makes the component menu in the integrated circuit interface more intuitive.

### PR DESCRIPTION
**Pull request by Watermelon914**
## About The Pull Request
Added a notice box to the component menu in the integrated circuit interface explaining how to link an integrated circuit to a component printer to allow for integrated circuit programming away from the component printer.

image

Last sentence in the above image has been removed. Integrated circuits will now update their static data once linked so closing and re-opening the UI is unnecessary.

## Why It's Good For The Game
It wasn't clear anywhere in game that you could link integrated circuits to component printers. I've run into players being completely unaware of this feature, so this should help to make it more clear that players don't need to print out components manually from the component printer and slap them into an integrated circuit each time.

## Changelog
:cl:
qol: Added a message to the component menu in the integrated circuit interface that explains how to link an integrated circuit to a component printer.
/:cl:

### CODE REVIEW

**Comment by MTandi** [comment]
Isn't this PR exactly the sign that component printing is not needed?

You add this message because you want it to be the main use case, right? Then it means that they never touch the circuit fabricator. Why not just link this directly to ore silo or lathe? What's the purpose of circuit fabricator now?

And how does this make sense? Are the components teleporting to the circuit via bluespace? Why does it take materials? 

Imagine that instead of this message, you'll just allow to add components without any links or usage of materials. What will be different? 

This is not a good example of lows that are needed to have highs. This is just a pain in the ass you need to overcome every time to start programming.

**Comment by Watermelon914** [comment]
> Isn't this PR exactly the sign that component printing is not needed?
> 
> You add this message because you want it to be the main use case, right? Then it means that they never touch the circuit fabricator. Why not just link this directly to ore silo or lathe? What's the purpose of circuit fabricator now?
> 
> And how does this make sense? Are the components teleporting to the circuit via bluespace? Why does it take materials?
> 
> Imagine that instead of this message, you'll just allow to add components without any links or usage of materials. What will be different?
> 
> This is not a good example of lows that are needed to have highs. This is just a pain in the ass you need to overcome every time to start programming.

to act as the research vessel and the machine to print shells and usb cables? component printers don't only vend components.

**Comment by Watermelon914** [comment]
> And how does this make sense? Are the components teleporting to the circuit via bluespace? Why does it take materials?

If that's how you want to explain it to yourself.

**Comment by Elli-Skala** [comment]
I you don't how many circuits I made before i learned this was a thing, it's. A obscure and mostly unlisted mechanic

**Comment by Watermelon914** [comment]
> I you don't how many circuits I made before i learned this was a thing, it's. A obscure and mostly unlisted mechanic

Yeah, it wasn't really intuitive, this makes it a lot, lot more clear for newer players.

**Comment by Vect0r2** [comment]
No! My aura of exclusivity and circuit wizardry, gone, and for what? Making it easy for those NEW PLAYERS to learn the game?

File: tgui/packages/tgui/interfaces/IntegratedCircuit/ComponentMenu.js [hunk 1]
```diff
@@ -133,6 +132,19 @@ export class ComponentMenu extends Component {
...
+              {trueLength === 0 && (
+                <Stack.Item mt={1} fontSize={1}>
+                  <NoticeBox info>
+                    You can hit this integrated circuit onto a component printer
+                    to link it so that you&apos;re able to remotely create and
+                    add components to this circuit.
+                    <br />
+                    <br />
+                    Close and re-open the integrated circuit interface once
+                    linked to fully complete the linking.
```
Comment (JohnFulpWillard) [inline tgui/packages/tgui/interfaces/IntegratedCircuit/ComponentMenu.js:144]: Why is this needed? Will refreshing static data fix it so this won't be necessary?

Comment (Watermelon914) [reply tgui/packages/tgui/interfaces/IntegratedCircuit/ComponentMenu.js:144]: yeah, if static data is refreshed

**Comment by MTandi** [comment]
>What you're trying to suggest and push for is that there shouldn't be resource limitations to circuits, which just shouldn't be the case.
> Linking is inconsequential compared to the things you are pushing for.

I'm asking to increase the price of the base circuit to whatever you feel should be average, and remove circuits as items and linking. So that people won't run out of materials in the middle of programming. You can leave the average material consumption unchanged, but people won't have to be distracted.

**Comment by Watermelon914** [comment]
> I'm asking to increase the price of the base circuit to whatever you feel should be average, and remove circuits as items and linking. So that people won't run out of materials in the middle of programming. You can leave the average material consumption unchanged, but people won't have to be distracted.

Complications when completing projects will happen all the time, and if resource deficit is one of your complications during programming, then resolving that by yelling at miners or finding ways to make materials is your only option. There are various other distractions you can run into when programming circuits, such as power drains, capacity limits, etc. These limitations exist to both balance out circuits and to give you a problem that you need to work around.

Something I've wanted to work on is introducing the concept of ghost components which players can use to plan out their circuits so that players can still work on their circuits when certain components or resources may not be available, but I don't think integrated circuits needs to take the extreme route and streamline itself like crazy.

**Comment by MTandi** [comment]
> > I'm asking to increase the price of the base circuit to whatever you feel should be average, and remove circuits as items and linking. So that people won't run out of materials in the middle of programming. You can leave the average material consumption unchanged, but people won't have to be distracted.
> 
> Complications when completing projects will happen all the time, and if resource deficit is one of your complications during programming, then resolving that by yelling at miners or finding ways to make materials is your only option. There are various other distractions you can run into when programming circuits, such as power drains, capacity limits, etc. These limitations exist to both balance out circuits and to give you a problem that you need to work around.
> 
> Something I've wanted to work on is introducing the concept of ghost components which players can use to plan out their circuits so that players can still work on their circuits when certain components or resources may not be available, but I don't think integrated circuits needs to take the extreme route and streamline itself like crazy.

See, there are already many distractions. If you remove this one, you will keep the balance needs, and the ghost components won't be needed. Why does this need to be so overcomplicated?

**Comment by Watermelon914** [comment]
> See, there are already many distractions. If you remove this one, you will keep the resource balance, and the ghost components won't be needed. Why does this need to be so overcomplicated?

Resource consumption isn't complicated.

**Comment by MTandi** [comment]
> > See, there are already many distractions. If you remove this one, you will keep the resource balance, and the ghost components won't be needed. Why does this need to be so overcomplicated?
> 
> Resource consumption isn't complicated. I'm not budging on this, I don't want to streamline circuits more than it already has been.

Ghost components will be complicated. And you can avoid this whatsoever if you make this simple change

**Comment by Mothblocks** [comment]
accessibility is for people who are disabled, this is just qol

but also this isn't "more intuitive", if it was intuitive you wouldn't need two sentences to explain it :P

**Comment by Watermelon914** [comment]
> accessibility is for people who are disabled, this is just qol
> 
> but also this isn't "more intuitive", if it was intuitive you wouldn't need two sentences to explain it :P

fair enough
